packages/jellyfish-testing/src/testingwrapper.ts

```diff
@@ -0,0 +1,57 @@
+import { RegTestFoundationKeys } from '@defichain/jellyfish-network'
+import { ContainerGroup, MasterNodeRegTestContainer, RegTestContainer } from '@defichain/testcontainers'
+import { Testing, TestingGroup, NonMNTesting } from '.'
+
+type InitMNRegContainerFn = (index: number) => MasterNodeRegTestContainer
+function defaultInitMNRegContainer (index: number): MasterNodeRegTestContainer {
+  return new MasterNodeRegTestContainer(RegTestFoundationKeys[index])
+}
+
+type InitNonMNRegContainerFn = (index: number) => RegTestContainer
+function defaultInitNonMNRegContainer (index: number): RegTestContainer {
+  return new RegTestContainer()
+}
+export const TestingWrapper = new (class TestingWrapperFactory {
+  create (): Testing
+  create (n: 0 | 1): Testing
+  create (n: number): TestingGroup
+  create (n: 0 | 1, init: InitMNRegContainerFn): Testing
+  create (n: number, init: InitMNRegContainerFn): TestingGroup
+
+  create (n?: number, init: InitMNRegContainerFn = defaultInitMNRegContainer): Testing | TestingGroup {
+    if (n === undefined || n <= 1) {
+      return Testing.create(init(0))
+    }
+
+    return TestingGroup.create(n, init)
+  }
+
+  createNonMN (): NonMNTesting
```

Comment:
Wanted to start a quick discussion on naming since it will be harder to change later once this starts spreading into our testing structures.

From initial reading `NonMN` twists the tongue a little. Wondering if perhaps we want to create a proper name for a reg container that is not a MasterNode. I was thinking perhaps PartialNode?

And then since we are creating individual methods with expected outcomes perhaps the wrapper exposes.

`createMasterNode` and `createPartialNode`

It creates a little more uniformity in naming and I think more clear in their meaning, thoughts?